File: java/src/main/java/com/meteroid/models/BatchJobStatus.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum BatchJobStatus implements ToQueryParam {
    PENDING("PENDING"),
    CHUNKING("CHUNKING"),
    PROCESSING("PROCESSING"),
    COMPLETED("COMPLETED"),
    COMPLETED_WITH_ERRORS("COMPLETED_WITH_ERRORS"),
    FAILED("FAILED"),
    CANCELLED("CANCELLED");
    private final String value;

    BatchJobStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: python/meteroid/models/batch_job_status.py
# this file is @generated
import enum


class BatchJobStatus(str, enum.Enum):
    PENDING = "PENDING"
    CHUNKING = "CHUNKING"
    PROCESSING = "PROCESSING"
    COMPLETED = "COMPLETED"
    COMPLETED_WITH_ERRORS = "COMPLETED_WITH_ERRORS"
    FAILED = "FAILED"
    CANCELLED = "CANCELLED"

    def __str__(self) -> str:
        return str(self.value)
